.travis.yml

```diff
@@ -4,38 +4,128 @@ notifications:
   email: false
 language: node_js
 node_js:
-  - "10"
   - "8"
-  - "6"
-jdk:
-  - oraclejdk9
-  - openjdk8
-env:
-  - MAVEN_VERSION=3.6.1 JAVA_VERSION=openjdk8
-  - MAVEN_VERSION=3.6.1 JAVA_VERSION=oraclejdk9
 
-  - MAVEN_VERSION=3.3.9 JAVA_VERSION=oraclejdk9
-  - MAVEN_VERSION=3.3.9 JAVA_VERSION=openjdk8
+cache:
+  directories:
+    - node_modules
+    - $HOME/.m2
 
 before_install:
-  # sdkman helps to install different gradle versions
+  # sdkman helps to install different maven versions
   - curl -s "https://get.sdkman.io" | bash
   - source "$HOME/.sdkman/bin/sdkman-init.sh"
   - echo sdkman_auto_answer=true > ~/.sdkman/etc/config
-cache:
-  directories:
-    - node_modules
-script:
-  - jdk_switcher use $JAVA_VERSION
-  - sdk install maven $MAVEN_VERSION
-  - npm test
+  # install jdk is needed for JDK10+
+  - wget https://github.com/sormuras/bach/raw/master/install-jdk.sh
+
 jobs:
   include:
+    # test MAVEN_VERSION=3.6.1
+    - stage: Node versions + actual Maven version
+      node_js: 6
+      env: MAVEN_VERSION=3.6.1 JDK=8
+      script:
+        - jdk_switcher use openjdk8
+        - sdk install maven $MAVEN_VERSION
+        - java -version
+        - npm test
+    - stage: Node versions + actual Maven version
```

Comment:
And here doe every maven version we already support: `3.6.1` and `3.3.9` tun these against most java versions but no need for a full matrix. Just so that we know the later maven version are working well on later java. Example form our gradle plugin:
<img width="1096" alt="Screen Shot 2019-08-07 at 12 11 00" src="https://user-images.githubusercontent.com/2911613/62618571-77abba80-b90c-11e9-9dcc-54c27e17419f.png">

Reply:
Should be done. Check the latest commit